src/java.desktop/macosx/classes/com/apple/laf/AquaComboBoxButton.java

```diff
@@ -154,6 +156,11 @@ public void paintComponent(final Graphics g) {
             } else {
                 painter.state.set(isSquare ? Widget.BUTTON_POP_UP_SQUARE : Widget.BUTTON_POP_UP);
             }
+
+            AccessibleContext ac = this.getAccessibleContext();
+            if (ac != null && (comboBox.getSelectedItem() != null)) {
+                ac.setAccessibleName(comboBox.getSelectedItem().toString());
```

Comment:
Why are you using the currently selected element for the button name and not a specific name? Whether it will sound right when using a screen reader. Wouldn't this be misleading to a blind user? Especially since it will only change when this method is called.

Reply:
> Why are you using the currently selected element for the button name and not a specific name?

As it is required to show the magnified text for the selected item in JComboBox. Setting it to some specific name always results into that name irrespective of selected item in JCombobox.

> Whether it will sound right when using a screen reader. Wouldn't this be misleading to a blind user?

I checked with VO announcement with or without this chunk of code, it is same in both case. 

> Especially since it will only change when this method is called.

This method is called whenever the current selection is changed so accessible name is set to correct item.

Reply:
JSON with enumeration of fields is also a text description of the object.
But if you checked that there is no garbage, then good.